client/src/containers/App/App.js

```diff
@@ -47,57 +60,143 @@ class App extends Component {
+            )}
+            {this.state.admin ? (
+              <Route
+                exact
+                path="/admin/lessons/edit/:lessonId"
+                component={EditLesson}
+              />
+            ) : (
+              <Route
+                exact
+                path="/admin/lessons/edit/:lessonId"
+                component={IsntAdmin}
+              />
+            )}
+            {this.state.admin ? (
+              <Route
+                exact
+                path="/admin/lessons/:lessonId/topics"
+                component={ViewTopics}
+              />
+            ) : (
+              <Route
+                exact
+                path="/admin/lessons/:lessonId/topics"
+                component={IsntAdmin}
+              />
+            )}
+            {this.state.admin ? (
+              <Route path="/admin/topics/add" component={AddTopics} />
+            ) : (
+              <Route path="/admin/topics/add" component={IsntAdmin} />
+            )}
+            {this.state.admin ? (
+              <Route
+                path="/admin/topics/edit/:topicId"
+                component={EditTopics}
+              />
+            ) : (
+              <Route path="/admin/topics/edit/:topicId" component={IsntAdmin} />
+            )}
+            {this.state.admin ? (
+              <Route exact path="/admin/users" component={Users} />
+            ) : (
+              <Route exact path="/admin/users" component={IsntAdmin} />
+            )}
+            {this.state.admin ? (
+              <Route
+                path="/admin/users/assign-role/:userId"
+                component={AssignUserRole}
+              />
+            ) : (
+              <Route
+                path="/admin/users/assign-role/:userId"
+                component={IsntAdmin}
+              />
+            )}
             <Route
               exact
               path="/lesson/:lessonId"
               component={ViewStudentTopics}
             />
-            <Route path="/admin/topics/add" component={AddTopics} />
-            <Route path="/admin/topics/edit/:topicId" component={EditTopics} />
-            <Route exact path="/admin/users" component={Users} />
-            <Route
-              path="/admin/users/assign-role/:userId"
-              component={AssignUserRole}
-            />
             <Route path="/user/profile/edit" component={EditUser} />
-            <Route
-              exact
-              path="/admin/courses/:courseId"
-              component={AssignCourseToStudent}
-            />
+            {this.state.admin ? (
+              <Route
+                exact
+                path="/admin/courses/:courseId"
+                component={AssignCourseToStudent}
+              />
+            ) : (
+              <Route
+                exact
+                path="/admin/courses/:courseId"
+                component={IsntAdmin}
+              />
+            )}
```

Comment:
Same comment here. Can you group all the routes of admin together and so just have one condition with this.state.admin?